File: 01-Assignment-02-02-2026/Using_C/program_21.c
/* Program 21: Implement Linear Search */

#include <stdio.h>

int LinearSearch(int array[], int value, int size) {
    for (int i = 0; i < size; i++) {
        if (array[i] == value)
            return i;
    }
    return -1;
}

int main() {
    int n;
    printf("Enter Length Of Your Array: ");
    scanf("%d", &n);

    int array[n];

    for (int i = 0; i < n; i++) {
        printf("Enter your No.%d : ", i + 1);
        scanf("%d", &array[i]);
    }

    int choice;
    while (1) {
        printf("\nWanna find a Number from your array?\nYes = 1\nNo = 0\n");
        scanf("%d", &choice);

        switch (choice) {
        case 1: {
            int x, index;
            printf("Enter value to search: ");
            scanf("%d", &x);

            index = LinearSearch(array, x, n);

            if (index != -1)
                printf("Element found at index %d\n", index);
            else
                printf("Element not found\n");
            break;
        }
        case 0:
            return 0;
        default:
            printf("Invalid choice\n");
        }
    }
}

File: 01-Assignment-02-02-2026/Using_Java/Program21.java
/*
Program 21
Java program to implement linear search
*/

import java.util.Scanner;

public class Program21 {
    public static int linearSearch(double[] nums,double target,int size){

        for(int i = 0;i < size;i++){
            if (nums[i] == target)
                return i;
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter size of the array : ");
        int size = scanner.nextInt();
        
        if (size < 0)
            System.out.println("Invalid Size");
        else{
            double[] array = new double[size];
            System.out.println("Lets Fill out the array : \n");
            for(int i = 0; i <  size;i++){
                System.out.print("Item No." + (i+1) + " : ");
                array[i] = scanner.nextDouble();
            }

            System.out.print("Which Item You want to search ? : ");
            double query = scanner.nextDouble();
            int res = linearSearch(array, query, size);
            if (res != -1){
                System.out.println(query + " is at index = " + res);
            }
            else{
                System.out.println("Item not found");
            }
            scanner.close();
        }

        
    }

}
